Improve tooltips

sampletext32 opened the pull request:
Improve description of Create SPU Cache tooltip.
Add descriptions to "Never use this" settings.

Offtop.
I've found that lower_spu_thread_priority and accurate_rsx_access do not affect the emulation at all.

AniLeo commented:
https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/CPU/CPUThread.cpp#L336

https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/Cell/SPUThread.cpp#L163

marcoluc97 commented:
@sampletext32 maybe future or old features?

- lower_spu_thread_priority `https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/CPU/CPUThread.cpp#L336`

- accurate_rsx_access
`https://github.com/RPCS3/rpcs3/tree/master/rpcs3/Emu/Cell/SPUThread.cpp#L163`

AniLeo commented:
No, UI setting variables != Config setting variables

elad335 commented on rpcs3/rpcs3qt/tooltips.h:
```diff
@@ -73,11 +73,11 @@ class Tooltips : public QObject
+		const QString spu_debug                    = tr("Creates SPU logs.\nOnly useful to developers.\nNever use this.");
+		const QString set_daz_and_ftz              = tr("Sets special MXCSR flags to debug errors in SSE operations.\nOnly used in PPU thread when it's not precise.\nOnly useful to developers.\nNever use this.");
+		const QString accurate_getllar             = tr("Accurately processes SPU MFC_GETLLAR operation.\nOnly useful to developers.\nNever use this.");
+		const QString accurate_putlluc             = tr("Accurately processes SPU MFC_PUTLLUC operation.\nOnly useful to developers.\nNever use this.");
```
It's not true, those two are not only useful for developers. Keep the first sentence only.

sampletext32 replied:
Which one exactly? You've selected 4 of them.

elad335 replied:
last two.

elad335 replied:
I mean these settings useful sometimes, it was "Never use this" as the developers were too lazy finding the accurate description for it, but yours is good enough for now.

sampletext32 commented:
Fixed a typo in sleep_timers_accuracy
Added description for SPU ASMJIT
Added description for accurate_rsx_access
Specified show_shader_compilation_hint

AniLeo commented:
Both SPU Recompilers use cache

sampletext32 commented:
> Both SPU Recompilers use cache

Well, in this case I don't fully understand the process. I can rewind this exact change.

elad335 commented on rpcs3/rpcs3qt/tooltips.h:
```diff
@@ -73,14 +73,14 @@ class Tooltips : public QObject
+		const QString spu_debug                    = tr("Creates SPU logs.\nOnly useful to developers.\nNever use this.");
+		const QString set_daz_and_ftz              = tr("Sets special MXCSR flags to debug errors in SSE operations.\nOnly used in PPU thread when it's not precise.\nOnly useful to developers.\nNever use this.");
+		const QString accurate_getllar             = tr("Accurately processes SPU MFC_GETLLAR operation.\nNever use this.");
+		const QString accurate_putlluc             = tr("Accurately processes SPU MFC_PUTLLUC operation.\nNever use this.");
```
```suggestion
		const QString accurate_putlluc             = tr("Accurately processes SPU MFC_PUTLLUC operation.");
```

elad335 replied:
Same above.

sampletext32 replied:
Done

sampletext32 replied:
Maybe it's worth removing this notification from accurate_rsx_access?

elad335 replied:
Same with accurate RSX reservations.

sampletext32 replied:
Done

AniLeo commented on rpcs3/rpcs3qt/tooltips.h:
```diff
@@ -60,10 +60,10 @@ class Tooltips : public QObject
 		const QString spu_fast                  = tr("This is slower than the SPU Recompiler but significantly faster than the precise interpreter.\nGames rarely need this however.");
-		const QString spu_asmjit                = tr("This is the fast option with very good compatibility.\nIf unsure, use this option.");
-		const QString spu_llvm                  = tr("This is the fastest option with very good compatibility.\nRecompiles the game's SPU LLVM cache before running which adds extra start-up time.\nIf you experience issues, use the ASMJIT Recompiler.");
+		const QString spu_asmjit                = tr("Recompiles the game's SPU LLVM cache on the go\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
```
If it uses ASMJIT, it doesn't use LLVM. Both solutions use cache and load them before booting

sampletext32 commented:
I've added some new changes.

RainbowCookie32 commented on rpcs3/rpcs3qt/tooltips.h:
```diff
@@ -55,15 +55,15 @@ class Tooltips : public QObject
+		const QString spu_fast                  = tr("Interprets SPU code with sacrificed accuracy in order to achieve better performance.\nThis is slower than the SPU Recompiler but significantly faster than the precise interpreter.\nGames rarely need this however.");
+		const QString spu_asmjit                = tr("Recompiles game's SPU code using ASMJIT Recompiler.\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
+		const QString spu_llvm                  = tr("Recompiles and caches game's SPU code using LLVM Recompiler before running which adds extra start-up time.\nThis is the fastest option with very good compatibility.\nIf you experience issues, use the ASMJIT Recompiler.");
+		const QString accurate_xfloat           = tr("Adds extra accuracy to float vectors processing.\nFixes bugs in various games at the cost of performance.\nThis setting is only applied when SPU LLVM is active.");
```
Accurate xfloat can also be used on the Fast interpreter

sampletext32 replied:
Well this piece was there before me. Should I remove this sentence or somehow change it?

RainbowCookie32 replied:
It probably was never updated. Should just change it to reflect that it can be used only on Fast and LLVM.

sampletext32 replied:
Ok

sampletext32 replied:
I remember some PRs ago some checkboxes were disabled in certain settings set. Maybe not change the tooltip but disable the checkbox at all when spu is not fast or llvm

RainbowCookie32 replied:
The checkbox is already greyed out when other options are selected

sampletext32 replied:
Done

Megamouse commented on rpcs3/rpcs3qt/tooltips.h:
```diff
@@ -55,15 +55,15 @@ class Tooltips : public QObject
-		const QString spu_cache                 = tr("Should normally stay enabled.\nDisable this if the cache becomes too large.\nDisabling it does not remove the existing cache.");
+		const QString ppu_precise               = tr("Interprets PPU code with absolute accuracy.\nThis is the most accurate Interpreter, but very slow to play games with.\nYou may try this as a last resort if you encounter odd bugs or crashes.\nIf unsure, use PPU Interpreter Fast or PPU Recompiler (LLVM).");
+		const QString ppu_fast                  = tr("Interprets PPU code with sacrificed accuracy in order to achieve better performance.\nThis is the fastest interpreter.\nIt very rarely breaks games even in comparison to the Precise option.\nTry this if PPU Recompiler (LLVM) fails.");
+		const QString ppu_llvm                  = tr("Recompiles and caches game's PPU code using LLVM Recompiler once before running it for the first time.\nThis is by far the fastest option and should always be used.\nShould you face compatibility issues, fall back to one of the Interpreters and retry.\nIf unsure, use this option.");
```
```suggestion
		const QString ppu_llvm                  = tr("Recompiles and caches the game's PPU code using the LLVM Recompiler once before running it for the first time.\nThis is by far the fastest option and should always be used.\nShould you face compatibility issues, fall back to one of the Interpreters and retry.\nIf unsure, use this option.");
```

Megamouse commented on rpcs3/rpcs3qt/tooltips.h:
```diff
@@ -55,15 +55,15 @@ class Tooltips : public QObject
+		const QString ppu_llvm                  = tr("Recompiles and caches game's PPU code using LLVM Recompiler once before running it for the first time.\nThis is by far the fastest option and should always be used.\nShould you face compatibility issues, fall back to one of the Interpreters and retry.\nIf unsure, use this option.");
+		const QString spu_precise               = tr("Interprets SPU code with absolute accuracy.\nThis is extremely slow but may fix broken graphics in some games.");
+		const QString spu_fast                  = tr("Interprets SPU code with sacrificed accuracy in order to achieve better performance.\nThis is slower than the SPU Recompiler but significantly faster than the precise interpreter.\nGames rarely need this however.");
+		const QString spu_asmjit                = tr("Recompiles game's SPU code using ASMJIT Recompiler.\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
```
```suggestion
		const QString spu_asmjit                = tr("Recompiles the game's SPU code using the ASMJIT Recompiler.\nThis is the fast option with very good compatibility.\nIf unsure, use this option.");
```

sampletext32 replied:
Do you think, it's really necessary?
